src/configuration-store/hybrid.store.ts

```diff
@@ -0,0 +1,59 @@
+import { logger } from '../application-logger';
+
+import { IAsyncStore, IConfigurationStore, ISyncStore } from './configuration-store';
+
+export class HybridConfigurationStore<T> implements IConfigurationStore<T> {
+  servingStore: ISyncStore<T>;
+  persistentStore: IAsyncStore<T> | null;
+
+  constructor(servingStore: ISyncStore<T>, persistentStore: IAsyncStore<T> | null) {
+    this.servingStore = servingStore;
+    this.persistentStore = persistentStore;
+  }
+
+  /**
+   * Initialize the configuration store by loading the entries from the persistent store into the serving store.
+   */
+  async init(): Promise<void> {
+    if (this.persistentStore) {
```

Comment:
I think here the guard/bouncer pattern would be best; makes it clear nothing happens and un-nests the rest of the code one level.
```ts
if (!this.persistentStore) {
  return;
}
```